atintegrators/atimplib.c

```diff
@@ -352,37 +331,29 @@ static void compute_kicks_longres(int nslice,int nbunch,int nturns, double *turn
             }
         }
     }
-    
-    vbeamk[0] /= totalW;
-    vbeamk[1] /= totalW;
-    
-    for(i=0;i<nbunch;i++){
-        totalW = 0.0;
-        loopstart = nslice*nbunch*(nturns-1) + i*nslice;
-        loopend = loopstart + nslice;
-        for(ii=loopstart;ii<loopend;ii++){
-            totalW += turnhistoryW[ii];
-        }
-        
-        double vr = vbr[i]/totalW;
-        double vi = vbi[i]/totalW;
-        vbr[i] = sqrt(vr*vr+vi*vi); 
-        vbi[i] = atan2(vi,vr);
-    }
-
 
     #ifdef MPI
     MPI_Allreduce(MPI_IN_PLACE,kz,nslice*nbunch,MPI_DOUBLE,MPI_SUM,MPI_COMM_WORLD);
     MPI_Allreduce(MPI_IN_PLACE,vbeamk,2,MPI_DOUBLE,MPI_SUM,MPI_COMM_WORLD);
     MPI_Allreduce(MPI_IN_PLACE,vbr,nbunch,MPI_DOUBLE,MPI_SUM,MPI_COMM_WORLD);
     MPI_Allreduce(MPI_IN_PLACE,vbi,nbunch,MPI_DOUBLE,MPI_SUM,MPI_COMM_WORLD);
+    MPI_Allreduce(MPI_IN_PLACE,&totalW,1,MPI_DOUBLE,MPI_SUM,MPI_COMM_WORLD);
+    MPI_Allreduce(MPI_IN_PLACE,totalWb,nbunch,MPI_DOUBLE,MPI_SUM,MPI_COMM_WORLD);
     MPI_Barrier(MPI_COMM_WORLD);
     #endif
 
-    vba = sqrt(vbeamk[0]*vbeamk[0]+vbeamk[1]*vbeamk[1]);   
-    vbp = atan2(vbeamk[1],vbeamk[0]);
+    vba = sqrt(vbeamk[0]*vbeamk[0]+vbeamk[1]*vbeamk[1])/totalW;
+    vbp = atan2(vbeamk[1],vbeamk[0])/totalW;
```

Comment:
here is the bug.

In the old script, I just divided vbeamk[0] and vbeamk[1] by totalW which is fine as it is real and imaginary voltage.

Line 345 is fine, but line 346 should not have the /totalW as anyway atan2 is  ratio so makes no sense

Reply:
Good catch in fact the phase should not be normalized